packages/idempotency/src/persistence/DynamoDBPersistenceLayer.ts

```diff
@@ -196,13 +197,26 @@ class DynamoDBPersistenceLayer extends BasePersistenceLayer {
             ':inprogress': IdempotencyRecordStatus.INPROGRESS,
           }),
           ConditionExpression: conditionExpression,
+          ReturnValuesOnConditionCheckFailure: 'ALL_OLD',
         })
       );
     } catch (error) {
       if (error instanceof DynamoDBServiceException) {
-        if (error.name === 'ConditionalCheckFailedException') {
+        if (error instanceof ConditionalCheckFailedException) {
+          if (!error.Item) {
+            throw new Error('Item is undefined');
+          }
+          const Item = unmarshall(error.Item);
```

Comment:
Please use small case for all variable names: `item` instead of `Item`

Reply:
this has been fixed in my recent commit.